src/legacy/core_plugins/visualizations/public/np_ready/public/legacy/build_pipeline.ts

```diff
@@ -59,7 +59,12 @@ export interface Schemas {
   [key: string]: any[] | undefined;
 }
 
-type buildVisFunction = (visState: VisState, schemas: Schemas, uiState: any) => string;
+type buildVisFunction = (
+  visState: VisState,
+  schemas: Schemas,
+  uiState: any,
+  meta?: { savedObjectId?: string }
```

Comment:
Maybe we want to make a TODO here or directly create an issue, that we remove this again once we enable hard validation?

Reply:
https://github.com/elastic/kibana/issues/54456